packages/flutter_test/lib/src/runner.dart

```diff
@@ -0,0 +1,322 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:isolate';
+import 'package:stack_trace/stack_trace.dart';
+import 'package:stream_channel/stream_channel.dart';
+import 'package:async/async.dart';
+import 'package:term_glyph/term_glyph.dart' as glyph;
+import 'package:test_api/src/backend/declarer.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/group.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/group_entry.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/state.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/message.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/invoker.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/live_test.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/metadata.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/stack_trace_formatter.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/suite.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/suite_platform.dart'; // ignore: implementation_imports
+import 'package:test_api/src/backend/test.dart'; // ignore: implementation_imports
+import 'package:test_api/src/util/remote_exception.dart'; // ignore: implementation_imports
+
+/// Capture any top-level errors (mostly lazy syntax errors, since other are
+/// caught below) and report them to the parent isolate.
+void catchIsolateErrors() {
+  final ReceivePort errorPort = ReceivePort();
+  // Treat errors non-fatal because otherwise they'll be double-printed.
+  Isolate.current.setErrorsFatal(false);
+  Isolate.current.addErrorListener(errorPort.sendPort);
+  errorPort.listen((dynamic message) {
+    // Masquerade as an IsolateSpawnException because that's what this would
+    // be if the error had been detected statically.
+    final IsolateSpawnException error = IsolateSpawnException(message[0]);
+    final Trace stackTrace =
+        message[1] == null ? Trace(const <Frame>[]) : Trace.parse(message[1]);
+    Zone.current.handleUncaughtError(error, stackTrace);
+  });
+}
+
+/// Returns a serialized test suite.
+StreamChannel<dynamic> serializeSuite(Function getMain(), {bool hidePrints = true, Future<dynamic> beforeLoad()}) {
+  return RemoteListener.start(getMain, hidePrints: hidePrints, beforeLoad: beforeLoad);
+}
+
+/// This is a listener.
+class RemoteListener {
```

Comment:
This logic is really specific to the internals of the runner. Sorry I didn't catch this the first time around but I think we can do a large clean up here if we do the following:

test_core/lib/src/runner/remote_listener.dart -> test_api/lib/src/remote_listener.dart
test_core/lib/src/runner/suite_channel_manager.dart -> test_api/lib/src/suite_channel_manager.dart

This way you won't have to fork this file and flutter_test still only depends on test_api. Thoughts?